src/components/utils/test/timer_test.cc

```diff
@@ -151,47 +141,61 @@ TEST_F(TimerTest, Start_Loop_3Calls) {
   test_lock.Release();
   test_timer.Stop();
 
+  EXPECT_FALSE(test_timer.is_running());
   EXPECT_EQ(loops_count, task->GetCallsCount());
 }
 
-TEST_F(TimerTest, Start_Runned_RunnedWithNewTimeout) {
+// {AKozoriz} : Disabled due correct realization of Timer
+// In case Start -> Immediately Stop | we have uncorrect behavior
+TEST_F(TimerTest, DISABLED_Start_Runned_RunnedWithNewTimeout) {
   // Preconditions
   timer::Timer test_timer(kTimerName, test_task_);
+  timeout_=1000u;
+  EXPECT_CALL(*test_task_, run()).Times(0);
   // Actions
   test_timer.Start(timeout_, single_shot_);
   // Expects
   ASSERT_EQ(timeout_, test_timer.timeout());
   ASSERT_TRUE(test_timer.is_running());
   // Actions
-  timeout_ = 1000u;
+  timeout_ = 200000u;
```

Comment:
@LuxoftAKutsan , why? we don`t care how big timeout can be putted to timer. And with this timeout we can easer find problem.